src/circuit_writer/mod.rs

```diff
@@ -163,12 +163,15 @@ impl<B: Backend> CircuitWriter<B> {
 
         // create public output
         if let Some(typ) = &function.sig.return_type {
-            if typ.kind != TyKind::Field {
-                unimplemented!();
+            match typ.kind {
+                TyKind::Field => {
+                    circuit_writer.add_public_outputs(1, typ.span);
+                }
+                TyKind::Array(_, len) => {
```

Comment:
I don't think this is correct as this won't work if you have `[T; len]` for `T` a different type than `Field`.

I think a quick fix here could be to match for `TyKind::Array(TyKind::Field, len)` or to actually recurse and figure out the real size (akin to the size_of function), but this would be out of the scope of this issue and perhaps a tad more complicated :o